### PR TITLE
Implement csv reader

### PR DESCRIPTION
**Pull request by acquamarin**

### CODE REVIEW

**Review by ray6080 — COMMENTED** [review]
As we discussed, we're going to revert changes on ValueVector capacity. Just push my comments here for a few files. Will continue later after the changes.

File: src/processor/operator/persistent/csv_reader.cpp [hunk 1]
```diff
@@ -0,0 +1,529 @@
...
+
+void BufferedCSVReader::Initialize() {
+    numPropertiesToCopy = 0;
+    for (auto& property : tableSchema->properties) {
+        if (property->getDataType()->getLogicalTypeID() != common::LogicalTypeID::SERIAL) {
+            numPropertiesToCopy++;
+        }
+    }
+    // TODO(Ziyi): should we wrap this fd using kuzu file handler?
+    fd = open(filePath.c_str(), O_RDONLY);
```
Comment (ray6080) [inline src/processor/operator/persistent/csv_reader.cpp:217]: We should reuse `FileUtils::openFile` if possible. The returned FileInfo will auto close file.

File: src/processor/operator/persistent/csv_reader.cpp [hunk 2]
```diff
@@ -0,0 +1,529 @@
...
+    }
+}
+
+BufferedCSVReader::BufferedCSVReader(const std::string& filePath,
+    common::CSVReaderConfig csvReaderConfig, catalog::TableSchema* tableSchema)
+    : BaseCSVReader{filePath, csvReaderConfig, tableSchema}, bufferSize{0}, position{0}, start{0} {
+    Initialize();
+}
+
+BufferedCSVReader::~BufferedCSVReader() {
```
Comment (ray6080) [inline src/processor/operator/persistent/csv_reader.cpp:203]: If you switch to FileUtils::openFile, we don't need to manually close the file.

**Review by ray6080 — APPROVED** [review]

File: src/storage/copier/node_group.cpp [hunk 3]
```diff
@@ -40,6 +40,10 @@ uint64_t NodeGroup::append(
     auto numValuesToAppendInChunk =
         std::min(numValuesToAppend, StorageConstants::NODE_GROUP_SIZE - numNodes);
     auto serialSkip = 0u;
+    auto originalVectorSize =
```
[the remark this answers is not in the document]
Comment (acquamarin) [reply src/storage/copier/node_group.cpp:43]: I think adding another parameter `numValuesToAppend` is a little wired. The valueVector already has a selectedSize which indicates the lenght of the valueVector.
BTW: ColumnChunk::append(common::ValueVector* vector, common::offset_t startPosInChunk) is also used by addProperty, the addProperty operator always append selectedSize number values.

**Pull request merged**